src/test/java/com/google/api/codegen/testsrc/protoannotations/library.proto

```diff
@@ -86,7 +86,7 @@ service LibraryService {
 
   // Deletes a shelf.
   rpc DeleteShelf(DeleteShelfRequest) returns (google.protobuf.Empty) {
-    option (google.api.http) = { delete: "/v1/{name=bookShelves/*}" };
+    option (google.api.http) = { delete: "/v1/bookShelves/{name}" };
```

Comment:
So we're supporting both `/{name=bookShelves/*}` and `/bookShelves/{name}`? It looks from https://google.aip.dev/127 that only the former is that standard. What am I overlooking?

Reply:
Both works, even if they are not listed in AIP127, so testing both.

Reply:
But if it's not in the AIP, we want to NOT support it. We don't want people annotating their gRPC APIs incorrectly, so we need to fail when they do. If the unapproved pattern is something that the converter is outputting into the synthetic protos, let's fix that. At the very least, could you file a bug in the converter repo and one in this repo, and add a TODO here to restrict to the supported pattern?

Reply:
This would require non-trivial unnecessary conversions in the converter which I really would like to avoid (one more thing to break potentially). This works out of the box in Java and PHP  and I'm pretty sure it will work in other langauges as well. If anything, I believe it is AIP description itself is incomplete, not the pattern used here is wrong. Note, that AIP does not really attempt to provide a full spec of how the template strings may look like.

Lets discuss it in person, I feel strong about not making the unnecessary conversion here (we are converting something simple to something more complicted, in a situation when it really looks unnecessary, to follow AIP which is not attempting to rigorously define how these template strings should look like).

Reply:
Btw, Looks like the annotation itself officially supports the plain syntax: https://github.com/googleapis/googleapis/blob/master/google/api/http.proto#L96

Reply:
Short version: OK, but add a note.

Long version:

In our discussion today we said that the HTTP annotation is independent of resource name formatting. It just says how to substitute the fields into an HTTP pattern, not how that pattern should look like,

I think that's wrong, though: from the AIP:
```
URIs must use the {foo=bar/*} syntax to represent a variable that should be populated in the request proto. When extracting a resource name, the variable must include the entire resource name, not just the ID component.
```

That said: it's OK for the monolith to accept this input for now. I don't think microgenerators should. I would put a TODO here noting that this is non-standard, even though we won't fix it in the monolith

(And the converter emits protos in the correct format, right?)

Reply:
To be clear, this isn't blocking this PR.

On the generation side, I checked and you're right, this pattern is intended to be permissible. The relevant AIP here is [AIP 4231](https://google.aip.dev/client-libraries/4231#complex-resource-id-path-segments) (Could you add a reference to that in the converter design doc, please?). I'll work with Luke to make the wording on AIP 127 less misleading (to me).